### PR TITLE
Limit organization size for auto upload

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: lib/foreman_inventory_upload.rb [hunk 1]
```diff
@@ -57,10 +57,14 @@ def self.upload_url
   end
 
   def self.slice_size
-    # for testing set ENV to 'https://ci.cloud.redhat.com/api/ingress/v1/upload'
     @slice_size ||= (ENV['SATELLITE_INVENTORY_SLICE_SIZE'] || '1000').to_i
   end
 
+  def self.max_org_size
+    # Set max amount of hosts per organization for automatic uploads
+    @max_org_size ||= (ENV['SATELLITE_INVENTORY_MAX_ORG_SIZE'] || '150000').to_i
```
Comment (adamruzicka) [inline lib/foreman_inventory_upload.rb:65]: The default can be an integer and `#to_i` is then just a noop

File: lib/foreman_inventory_upload/async/generate_all_reports_job.rb [hunk 2]
```diff
@@ -13,7 +13,13 @@ def perform
         organizations = Organization.unscoped.all
 
         organizations.map do |organization|
```
Comment (adamruzicka) [inline lib/foreman_inventory_upload/async/generate_all_reports_job.rb:15]: This previously returned an array of active jobs and now returns an array where elements can be either active jobs or `nil`s. Is the result used anywhere? If not, could we switch `#map` for `#each`?

**Review by adamruzicka — COMMENTED** [review]
Two nitpicks, otherwise lgtm

File: lib/foreman_inventory_upload/async/generate_all_reports_job.rb (hunk 2 above)
Comment (ShimShtein) [reply lib/foreman_inventory_upload/async/generate_all_reports_job.rb:15]: The method anyway was returning the `ensure` block. I think the `map` could be replaced with `each` anyway.

**Comment by ShimShtein** [comment]
@adamruzicka, ready for another round

File: lib/foreman_inventory_upload/async/generate_all_reports_job.rb (hunk 2 above)
Comment (adamruzicka) [reply lib/foreman_inventory_upload/async/generate_all_reports_job.rb:15]: I always thought `ensure` blocks get evaluated last, but do not alter the return value

Comment (ShimShtein) [reply lib/foreman_inventory_upload/async/generate_all_reports_job.rb:15]: Yep, you're right. `ensure` does not return value.

Well, removed `nil` values. Now the return value will filter out `nil` values.

**Comment by ShimShtein** [comment]
@adamruzicka, ready for another round.

**Review by adamruzicka — APPROVED** [review]

**Review by laviro — APPROVED** [review]
Well, good to know that someone uses Foreman with that amount of hosts :D

**Comment by laviro** [comment]
Thanks @ShimShtein @adamruzicka !

**Pull request merged**